### PR TITLE
show fixtures by gameweek in SwiftUI client

### PR DESCRIPTION
**Pull request by joreilly**
Hacked updates to SwiftUI code to allow browsing fixtures by game week (building on changes @HoinzeyBear made for Android client).  Had layout issues with those buttons inside `NavigationView` so put above for now.

Simulator Screen Shot - iPhone 12 Pro Max - 2022-09-18 at 10 59 04

### CODE REVIEW

**Comment by joreilly** [comment]
fyi @HoinzeyBear ....had to change type of `_gameweekToFixtureMap` a little (and associated code to create it) due to some issues I had with mapping to Swift code....think probably also good to have immutable types in cases like this if possible.

File: common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt [hunk 1]
```diff
@@ -120,46 +120,42 @@ class FantasyPremierLeagueRepository : KoinComponent {
...
+                                return@let gf
                             }
-
-                            return@let gf
                         }
-                    }
+                        //Build gameweek to fixture map
+                        _gameweekToFixtureMap.value = _fixtureList.value
+                            .filter { it.event != null }
+                            .groupBy { it.event!! }
```
Comment (joreilly) [inline common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt:158]: Must be cleaner way to do this with needed "!!"

Comment (HoinzeyBear) [reply common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt:158]: Looking at this it seems none of the fixtures in `fixtureList` have event == null. Presumably they all get lost where you check that they have a `kickOffTime`. 

So in that case we could remove the nullability from event on the `GameFixture `object which then removes that filter & double-bang ?

Though I'm not sure what those first few results from the fixtures api call are that have no kickoff time or event.

Comment (HoinzeyBear) [reply common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt:158]: [This MR](https://github.com/joreilly/FantasyPremierLeague/pull/44) does this.

Comment (joreilly) [reply common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt:158]: They have be fixtures that were cancelled/postponed but haven't been rescheduled yet

**Pull request merged**